### DISCUSSION TITLE
Weekly #6

### DISCUSSION THREAD

**Discussion by Ninetilt** [Ninetilt/Project-TDC#12] [General]
# **Übersicht**
Unser dritter Sprint ist gut ins Rollen gekommen und wir befinden uns jetzt schon mitten in der Entwicklung unserer TDC-App. Deshalb wollen wir euch diese Woche über unsere umgesetzten Features berichten, Probleme die bei dieser Umsetzung angefallen sind und euch unsere strukturellen Änderungen/Erweiterung unserer Entwicklung mit Git zeigen.

## **Feature-Entwicklung**
Ganze vier Features aus unseren User-Stories haben unsere fleißigen Entwickler diese Woche umgesetzt, die wir euch nun kurz vorstellen möchten.

### 1. Erstellung einer Todo-List
Dies ist relativ selbst erklärend. Aktuell wird das Erstellen einer neuen Liste ganz einfach über einen Button auf der Homepage gelöst, über den man dann zum nächsten Schritt kommt um dort auch den Titel einzugeben.

![homepage](https://github.com/user-attachments/assets/4c395e4f-b487-41c0-8bd1-81f200a73859) ![listcreated](https://github.com/user-attachments/assets/b6d49986-fe87-4250-972b-e0d47f0ee235)

### 2. Erstellen und Hinzufügen von Todo-Items
Direkt aufbauend auf der Erstellung einer Liste folgt das Erstellen der Aufgaben der Todo-Liste, sog. "Todo-Items". Über den Button "Add new Items" können Aufgaben eingegeben werden und mit der Höhe des Aufwands zum Erledigen dieses Tasks versehen werden (Aufwandbetitelung WIP).

![createitem](https://github.com/user-attachments/assets/f6ed8bb6-f2e3-4f27-95be-d21d773fedbc) ![itemscreated](https://github.com/user-attachments/assets/c3f0ac60-6f12-4c6b-93ba-cb7ff6137999)

### 3. Entfernen von Todo-Items über Backspace
Beim Erstellen neuer Todo-Items soll man auch schnell in der Lage sein fälschlich oder zu viel erstellte Items wieder zu löschen. Dies ist durch einen Backspace möglich. Dieses Feature hat eine besondere Herausforderung dargestellt, da .NET MAUI leider nicht über eine eingebaute "key-down"-Funktion verfügt und deswegen mit Blut, Schweiß und Tränen ein Workaround dafür gefunden werden musste.

### 4. Speichern und Abrufen von Listen
Über den Button "Save List" kann eine bearbeitete Liste gespeichert werden. Aktuell geschieht das über eine Erstellung einer .csv Datei worin die Daten der Liste hinterlegt, um diese danach auch wieder auslesen zu können. Das auslesen der .csv Datei funktioniert auch schon. Diese dann wieder in eine Liste zu überführen wurde aber noch nicht Implementiert. Die Lösung mit Hilfe von .csv Dateien ist nur temporär und wieder später über eine Datenbank geregelt.

![csv](https://github.com/user-attachments/assets/407d5346-642a-447a-a183-9b9d2ce546b3)

## Neues in Git
Hier in Git könnt ihr nun in einer [Readme](https://github.com/Ninetilt/Project-TDC/blob/dev/README.md) die wichtigsten Informationen zu unserem Projekt zusammengefasst finden. Zusätzlich haben wir für uns Developer jetzt auch genauer die Handhabung mit den verschiedenen Branches in der Entwicklung definiert, damit in der jetzt richtig gestarteten Entwicklung alles reibungslos ablaufen kann.

**Comment by Kartoffeleintopf** [comment]
Sieht gut aus, eure Vision der ToDo-App wird langsam Wirklichkeit.
Die Übergangslösung der Persistierung mit einer CSV-Datei finde ich interessant.
In unserem Team gab es auch bereits Überlegungen dieser Art, allerdings haben wir JSON bedacht.
Beim stöbern durch eure Readme fiel mir eure Branch Struktur auf.
In unserem eigenem Repo haben wir Branches nicht in einer einheitlichen Struktur benannt.
Ergo erachte ich eure dokumentierte Branch Struktur positiv und vielleicht kann man sich das mopsen.

> Euer TripAhead-Team (CN)

**Comment by SidMllr** [comment]
Es ist beeindruckend zu sehen, wie gut ihr im dritten Sprint Fortschritte gemacht habt, besonders mit den neuen Features wie dem CSV-Export und der in harter Arbeit erkämpften Lösung für die .NET MAUI-Problematik. Eure klare Branch-Struktur und Dokumentation sind vorbildlich und erleichtern bestimmt die Zusammenarbeit im Team.
Eure temporäre Lösung der CSV-Exports sieht sehr gelungen aus, auch unser Team wird in Zukunft eine Datenbank implementieren und davor vielleicht eine ähnliche Lösung brauchen, für die wir jetzt (dank euch) eine Idee haben!
Weiter so, euer Ansatz ist sehr inspirierend, wir sind gespannt die weitere Entwicklung der nächsten Wochen zu verfolgen.

Euer Project Haelth-Team
